flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingMetricCollector.java

```diff
@@ -343,8 +346,19 @@ private double computeLagGrowthRate(
         return (currentLag - lastLag) / timeDiff;
     }
 
+    private Map<JobVertexID, Map<String, FlinkMetric>> queryFilteredMetricNames(
+            Context ctx, JobTopology topology, boolean isStabilizing) {
+        try {
+            return queryFilteredMetricNames(ctx, topology);
+        } catch (MetricNotFoundException e) {
+            if (isStabilizing) {
+                throw new RecoverableException(e);
+            }
+            throw e;
+        }
+    }
+
     /** Query the available metric names for each job vertex. */
-    @SneakyThrows
     protected Map<JobVertexID, Map<String, FlinkMetric>> queryFilteredMetricNames(
             Context ctx, JobTopology topology) {
```

Comment:
nit: could these 2 `queryFilteredMetricNames` methods be inlined? The old `queryFilteredMetricNames` method is only called by new `queryFilteredMetricNames` method.

Reply:
I did not inline because this method is overridden in tests, e.g. `TestingScalingCollector`. Inlining the newly-added method into the protected method would change the behavior. I wanted to make sure that the newly added method is never overridden.

Reply:
thanks for the clarification, sounds make sense.